engine/runtime/src/main/scala/org/enso/interpreter/instrument/command/ImportSuggestionCmd.scala

```diff
@@ -0,0 +1,120 @@
+package org.enso.interpreter.instrument.command
+
+import org.enso.compiler.core.IR
+import org.enso.interpreter.instrument.execution.RuntimeContext
+import org.enso.interpreter.runtime.Module
+import org.enso.polyglot.runtime.Runtime.Api
+
+import scala.collection.mutable
+import scala.concurrent.{ExecutionContext, Future}
+
+/** A command that gathers info required for suggestion import.
+  *
+  * @param maybeRequestId an option with request id
+  * @param request a request for suggestion import
+  */
+final class ImportSuggestionCmd(
+  maybeRequestId: Option[Api.RequestId],
+  val request: Api.ImportSuggestionRequest
+) extends Command(maybeRequestId) {
+
+  import ImportSuggestionCmd._
+
+  /** Executes a request.
+    *
+    * @param ctx contains suppliers of services to perform a request
+    * @param ec execution context
+    */
+  override def execute(implicit
+    ctx: RuntimeContext,
+    ec: ExecutionContext
+  ): Future[Unit] = Future {
+    val suggestion = request.suggestion
+    reply(
+      Api.ImportSuggestionResponse(
+        suggestion.module,
+        suggestion.name,
+        findExports.sortBy(_.depth).map(_.export)
+      )
+    )
+  }
+
+  /** Find re-exports of the given symbol.
+    *
+    * @param ctx contains suppliers of services to perform a request
+    */
+  private def findExports(implicit ctx: RuntimeContext): Seq[ExportResult] = {
+    val suggestion = request.suggestion
+    val topScope =
+      ctx.executionService.getContext.getCompiler.context.getTopScope
+    @scala.annotation.tailrec
+    def go(
+      queue: mutable.Queue[ExportingModule],
+      builder: mutable.Builder[ExportResult, Vector[ExportResult]]
+    ): Vector[ExportResult] =
+      if (queue.isEmpty) {
+        builder.result()
+      } else {
+        val current = queue.dequeue()
```

Comment:
This duplicates a lot of logic from exports resolver unnecessarily, looking at the syntactic representation of exports for no reason. All of this is completely resolved in a compiled module, look at [BindingsMap](https://github.com/enso-org/enso/blob/bc8a22e27983d1b2a48809d329ce57576b5fa9dc/engine/runtime/src/main/scala/org/enso/compiler/data/BindingsMap.scala#L29:L33). It already contains all the information you need – in particular a map from string names to resolutions that would allow you to quickly query "does the module re-export a given symbol"?
Please use that structure instead, as it is much more resilient against future language semantics changes.